clients/mobile/src/database/MigrationManager.ts

```diff
@@ -0,0 +1,73 @@
+import SQLite from 'react-native-sqlite-storage'
+
+export type Migration = {
+  version: number
+  up: (db: SQLite.SQLiteDatabase) => Promise<void>
+  down?: (db: SQLite.SQLiteDatabase) => Promise<void>
+}
+
+class MigrationManager {
+  private db: SQLite.SQLiteDatabase
+  private migrations: Migration[]
+
+  constructor(db: SQLite.SQLiteDatabase, migrations: Migration[]) {
+    this.db = db
+    this.migrations = migrations.sort((a, b) => a.version - b.version)
+  }
+
+  private async getCurrentVersion(): Promise<number> {
+    return new Promise((resolve, reject) => {
+      this.db.transaction(tx => {
+        tx.executeSql(
+          'CREATE TABLE IF NOT EXISTS Version (key TEXT PRIMARY KEY, value TEXT);',
+          [],
+          () => {
+            tx.executeSql(
+              'SELECT value FROM Version WHERE key = "schema_version";',
+              [],
+              (_, result) => {
+                if (result.rows.length > 0) {
+                  resolve(parseInt(result.rows.item(0).value, 10))
+                } else {
+                  tx.executeSql(
+                    'INSERT INTO Version (key, value) VALUES ("schema_version", "1");',
```

Comment:
when version doesn't exist yet ,it should start with `0` , otherwise the `1.ts` migration will not run